### PR TITLE
Update introduction

### LINKED ISSUE #62: Write/update documentation
Tasks:

- [ ]  Write user guide
- [ ]  Write Developers guide

### PR DESCRIPTION
**Pull request by anjastrunk**
closes #62

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,182 +1,210 @@
...
-This then should be output as GX-compliant credential in JSON-LD for the Gaia-X
-catalogue.
+Gaia-X defines several level of compliance, each with a different trust level. gx-credential-generator supports the very basic level, called "Conformity" and is compliant Gaia-X Tagus release.
 
-### k8s
+### Cloud Service Provider
 
-We want to collect discoverable information from a k8s cloud,
-assuming that we have access to it.
+gx-credential generator outputs the following Gaia-X Credentials for a CSP in order to be Gaia-X compliant
```
Comment (markus-hentsch) [inline README.md:36]:
```suggestion
gx-credential-generator outputs the following Gaia-X Credentials for a CSP in order to be Gaia-X compliant
```

File: README.md [hunk 2]
```diff
@@ -1,182 +1,210 @@
...
+### OpenStack
 
-- Metadata
-- API Version
-- Nodes information
-- Pods information
+gx-credential-generator collects discoverable information from an OpenStack cloud, bundles them to Gaia-X Credentials and requests for compliance at GXDCH.
 
-#### K8s as-a-Service (KaaS) offering considerations
+Besides the Gaia-X Credentials of CSP of OpenStack cloud, with are required by Gaia-X, the follwing Gaia-X Credentials will be created:
```
Comment (markus-hentsch) [inline README.md:49]:
```suggestion
Besides the Gaia-X Credentials of a CSP of an OpenStack cloud, which are required by Gaia-X, the following Gaia-X Credentials will be created:
```

File: README.md [hunk 3]
```diff
@@ -1,182 +1,210 @@
...
-it, so users can use the GX Credentials to match it to app requirements.
+- Mandatory properties for IaaS Offering as an instance of Gaia-X `ServiceOffering`
+- Detailed description of OpenStack cloud as an instance of Gaia-X `VirtualMachineServiceOffering`
+- Compliance Credentials for OpenStack cloud issued by GXDCH Compliance Service as an instance of Gaia-X class `compliance`
 
-So the GX Credential for a k8s aaS solution would list possible options and
-ranges: What k8s versions are supported, what max number of workers,
-flavors, etc.? What services are optionally delivered (and supported)
-by the provider?
+To discover cloud properties, gx-credential generator requires access to OpenStack cloud as normal tenant user.
```
Comment (markus-hentsch) [inline README.md:55]:
```suggestion
To discover cloud properties, gx-credential-generator requires access to the OpenStack cloud as normal tenant user.
```

File: README.md [hunk 4]
```diff
@@ -1,182 +1,210 @@
...
 
-For KaaS, the option space really needs to be described.
-As of now, this can not be discovered, short of using external sources,
-like the IaaS GX Credential, the list of node images (osism minio), ...
+gx-credential-generator read the OpenStack catalog to collect
```
Comment (markus-hentsch) [inline README.md:57]:
```suggestion
gx-credential-generator queries the OpenStack API to collect
```

File: README.md [hunk 5]
```diff
@@ -1,182 +1,210 @@
...
+
+- public VM Images and their properties, such as operation system or hardware requirements
+- public Server Flavors, such as number and capability of virtual CPUs or size of root disk
+
+### k8s
```
Comment (markus-hentsch) [inline README.md:62]:
```suggestion
### Kubernetes (k8s)
```

File: README.md [hunk 6]
```diff
@@ -1,182 +1,210 @@
...
+
+#### 2. Run gx-credential-generator
+
+Create Gaia-X Credential for CSP without specifying a configuration file. This implies the default path at `/etc/gx-credential-generator/config.yaml`, which must exist:
+
+```commandline
+python3 -m generator.cli csp
+```
+
+Gaia-X terms and conditions are prompt and you are asked to agree to them. Type 'y' to agree or 'no' to dis-agree.
```
Comment (markus-hentsch) [inline README.md:97]:
```suggestion
Gaia-X terms and conditions are displayed and you are prompted to agree to them. Type 'y' to agree or 'n' to disagree.
```

File: README.md [hunk 7]
```diff
@@ -1,182 +1,210 @@
...
+
+**Note**: If you do not agree Gaia-X terms and conditions, process will be aborted and none Gaia-X credential is created.
```
Comment (markus-hentsch) [inline README.md:99]:
```suggestion
**Note**: If you do not agree Gaia-X terms and conditions, the process will be aborted and no Gaia-X credential is created.
```

File: README.md [hunk 8]
```diff
@@ -1,182 +1,210 @@
...
+
+Each Gaia-X Credential is serialized in [JSON-LD](https://json-ld.org/) and stored in a separate file prefixed as follows:
+
+- lp: Gaia-X Credential containing CSP's legal address and headquarter address
+- lrn: Gaia-X Credential for CSP's legal registration number issued by GXDCH Notary Service. gx-credential-generator reaches out to Notary Service by itself.
+- tandc: Gaia-X Terms and Conditions signed by CSP
+- vp_csp: Presentation of all Gaia-X Credentials to be sent to GXDCH Compliance Service to assert compliance
+- cs_csp: Compliance Credential for CSP as Gaia-X `LegalPerson` issued by GXDCH Compliance Service
+
+Gaia-X Credential files are placed in current working directory, be default. To change output directory use parameter `--out-dir`:
```
Comment (markus-hentsch) [inline README.md:109]:
```suggestion
Gaia-X Credential files are placed in the current working directory, by default. To change the output directory use the parameter `--out-dir`:
```

File: README.md [hunk 9]
```diff
@@ -1,182 +1,210 @@
...
+
+```commandline
+python3 -m generator.cli csp --out-dir=my-output-dir
+```
+
+Running the gx-credential-generator with a specified configuration file path use parameter `--config`:
```
Comment (markus-hentsch) [inline README.md:115]:
```suggestion
Running the gx-credential-generator with a specified configuration file path using the parameter `--config`:
```

File: README.md [hunk 10]
```diff
@@ -1,182 +1,210 @@
...
+
+```commandline
+python3 -m generator.cli csp --config=my-config.yaml
+```
+
+You can prevent to prompt Gaia-X terms and conditions using option `--auto-sign`. This implies you agree to them:
```
Comment (markus-hentsch) [inline README.md:121]:
```suggestion
You can avoid interactive prompt for Gaia-X terms and conditions agreement using the option `--auto-sign`. This implies you agree to them:
```

File: README.md [hunk 11]
```diff
@@ -1,182 +1,210 @@
...
+python3 -m generator.cli csp --auto-sign
+```
+
 ### OpenStack
 
-a. Create `clouds.yaml` configuration file
+#### 1. Create `clouds.yaml` configuration file
 
-- GX credential Generator requires access to OpenStack cluster as normal tenant
+gx-credential-generator requires access to OpenStack cluster as normal tenant
```
Comment (markus-hentsch) [inline README.md:131]:
```suggestion
gx-credential-generator requires access to the OpenStack API as a normal tenant
```

File: README.md [hunk 12]
```diff
@@ -1,182 +1,210 @@
...
   user and has to be configured with these user credentials to access your
   Openstack cloud. This is done
```
Comment (markus-hentsch) [inline README.md:133]:
```suggestion
  OpenStack cloud. This is done
```

File: README.md [hunk 13]
```diff
@@ -1,182 +1,210 @@
...
   using [clouds.yaml](https://docs.openstack.org/python-openstackclient/ussuri/configuration/index.html).
-  clouds.yaml is a yaml file containing several cloud configurations. Each
-  configuration is referred by name.
-- Make sure the following keys exist in our `clouds.yaml`.
-  - `auth.user_domain_name`
-  - `auth.project_domain_name`
-  - `region_name`
+  clouds.yaml is a yaml file containing several cloud configurations. Each configuration is referred by name.
```
Comment (markus-hentsch) [inline README.md:135]:
```suggestion
A `clouds.yaml` is a YAML file containing several cloud access configurations. Each configuration is referred to by name.
```

File: README.md [hunk 14]
```diff
@@ -1,182 +1,210 @@
...
 ### OpenStack
 
-a. Create `clouds.yaml` configuration file
+#### 1. Create `clouds.yaml` configuration file
 
-- GX credential Generator requires access to OpenStack cluster as normal tenant
+gx-credential-generator requires access to OpenStack cluster as normal tenant
   user and has to be configured with these user credentials to access your
   Openstack cloud. This is done
   using [clouds.yaml](https://docs.openstack.org/python-openstackclient/ussuri/configuration/index.html).
```
Comment (markus-hentsch) [inline README.md:134]:
```suggestion
  using [clouds.yaml](https://docs.openstack.org/python-openstackclient/latest/configuration/index.html).
```

File: README.md [hunk 15]
```diff
@@ -1,182 +1,210 @@
...
-  ./gx-cred-generator.py k8s
-  ```
-
-### Airflow Pipeline
-
-Start the gaiax-pipeline (deprecated)
-
-- To modify the airflow pipeline you have to touch the gaiax-pipeline.py file
-  inside the dags folder
+Create Gaia-X Credential for Openstack cloud
```
Comment (markus-hentsch) [inline README.md:151]:
```suggestion
Create Gaia-X Credential for an OpenStack cloud
```

File: README.md [hunk 16]
```diff
@@ -1,182 +1,210 @@
...
 
   ```bash
-  cd devops
-  docker-compose up -d
+  python3 -m generator.cli openstack <os-cloud>
   ```
```
Comment (markus-hentsch) [inline README.md:156]:
```suggestion

(`<os-cloud>` is a placeholder for the name of the desired entry in `clouds.yaml`)

```

File: README.md [hunk 17]
```diff
@@ -1,182 +1,210 @@
...
 
-### Validation
+As Gaia-X requires to define a provider for each published service offering, gx-credential-generator creates Gaia-X Credentials for CSP at very run for OpenStack cloud, too.
```
Comment (markus-hentsch) [inline README.md:157]:
```suggestion
As Gaia-X requires to define a provider for each published service offering, gx-credential-generator creates Gaia-X Credentials for the CSP at every run for an OpenStack cloud, too.
```

File: README.md [hunk 18]
```diff
@@ -1,182 +1,210 @@
...
 
-Generated GX Credentials could be validated against their schemas (shapes) by
-the
-simple SD validator script. Visit the `sd` directory and try to validate your
-generated SD. Find the examples in `sd` directory and do the validation as
-follows:
+Each Gaia-X Credential is serialized in [JSON-LD](https://json-ld.org/) and stored in a separate file. Credentials for CSP correspond to the ones generated by command `csp`. Credentials for OpenStack cloud are prefixed as follows:
```
Comment (markus-hentsch) [inline README.md:159]:
```suggestion
Each Gaia-X Credential is serialized in [JSON-LD](https://json-ld.org/) and stored in a separate file. Credentials for CSP correspond to the ones generated by the command `csp`. Credentials for OpenStack clouds are prefixed as follows:
```

File: README.md [hunk 19]
```diff
@@ -1,182 +1,210 @@
...
 
-Try to validate a minimal example against the latest GX shapes (feel free to
-remove some
-required attribute and check validation result):
+- so: Mandatory properties for OpenStack cloud
```
Comment (markus-hentsch) [inline README.md:161]:
```suggestion
- so: Mandatory properties for the OpenStack cloud
```

File: README.md [hunk 20]
```diff
@@ -1,182 +1,210 @@
...
+- vmso: Detailed description of OpenStack cloud
```
Comment (markus-hentsch) [inline README.md:162]:
```suggestion
- vmso: Detailed description of the OpenStack cloud
```

File: README.md [hunk 21]
```diff
@@ -1,182 +1,210 @@
...
+- vp_so: Presentation of all Gaia-X Credentials to be sent to GXDCH Compliance Service to assert compliance
+- cs_so: Compliance Credentials for OpenStack cloud as Gaia-X `ServiceOffering` issued by GXDCH Compliance Service
```
Comment (markus-hentsch) [inline README.md:164]:
```suggestion
- cs_so: Compliance Credentials for the OpenStack cloud as Gaia-X `ServiceOffering` issued by GXDCH Compliance Service
```

File: README.md [hunk 22]
```diff
@@ -1,182 +1,210 @@
...
 
-```bash
-./sd/validate.py sd/gx_service_offering_example.jsonld sd/gx_shapes_latest.ttl
-```
+### K8s
```
Comment (markus-hentsch) [inline README.md:166]:
```suggestion
### Kubernetes (k8s)
```

File: README.md [hunk 23]
```diff
@@ -1,182 +1,210 @@
...
-the [GX registry](https://registry.lab.gaia-x.eu/v1/api/trusted-shape-registry/v1/shapes/trustframework).
-GX Credential Generator **does not** create W3C
-complaint [Verifiable Credentials](https://www.w3.org/TR/vc-data-model/).
+## User Guide
 
-## Configuration
+### Configuration
 
-GX Credential generator is configured by `config.yaml`. The configuration
+gx-credential-generartor is configured by `config.yaml`. The configuration
```
Comment (markus-hentsch) [inline README.md:174]:
```suggestion
gx-credential-generator is configured by `config.yaml`. The configuration
```

File: README.md [hunk 24]
```diff
@@ -1,182 +1,210 @@
...
 
 ### Mandatory Attributes
 
-Gaia-X Credential schema dictates mandatory attributes for some class. If values
-for mandatory attributes can not be access from OpenStack or K8S cluster,
-default values are taken from configuration file in section `default`.
-Providers are able to change default values. In doing so, attribute values for *
-*ALL** instances of impacted cloud resource are modified.
+Gaia-X Credential schema dictates mandatory attributes for some class.
+If values for mandatory attributes can not be discovered from OpenStack or K8S cluster, default values are taken from configuration.
```
Comment (markus-hentsch) [inline README.md:184]:
```suggestion
If values for mandatory attributes cannot be discovered from the OpenStack cloud or Kubernetes cluster, default values are taken from the configuration.
```

File: README.md [hunk 25]
```diff
@@ -1,182 +1,210 @@
...
+- Enpoints to GXDCH services
 
 ### Mandatory Attributes
 
-Gaia-X Credential schema dictates mandatory attributes for some class. If values
-for mandatory attributes can not be access from OpenStack or K8S cluster,
-default values are taken from configuration file in section `default`.
-Providers are able to change default values. In doing so, attribute values for *
-*ALL** instances of impacted cloud resource are modified.
+Gaia-X Credential schema dictates mandatory attributes for some class.
```
Comment (markus-hentsch) [inline README.md:183]:
```suggestion
The Gaia-X Credential schema dictates mandatory attributes for some classes.
```

File: README.md [hunk 26]
```diff
@@ -1,182 +1,210 @@
...
 
-#### CopyrightOwner, License and ResourcePolicy of VM images
+#### CopyrightOwner, License and ResourcePolicy of VM images and Operating System
 
 `copyrightOwner`, `license` and `resourcePolicy` are mandatory attributes for VM
 Images and their operating systems. As these values are not accessible from
 OpenStack cloud, default values are used. The values for operating system are
-defined in the section `operating system` with one subsection for each operating
+defined in the section ``software resources`` with one subsection for each operating
 system. Operating systems are referred by name, e.g. for Alpine Linux:
```
Comment (markus-hentsch) [inline README.md:194]:
```suggestion
system. Operating systems are referenced by name, e.g. for Alpine Linux:
```

File: README.md [hunk 27]
```diff
@@ -196,11 +224,17 @@ cloud resources:
         - https://www.example.org
 ```
 
-### Cryptographical Material
+### Prerequisites to create and sign Gaia-X Credential
+
+gx-credential-generator creates Gaia-X Credentials, which refer to [W3C Verifiable Credentials](https://www.w3.org/TR/vc-data-model-2.0/). Verifiable Credentials requires a proof, e.g. a digital signature of credential's issuer. Therefore some settings, e.g. a private key to sign, are required and defined in section `Credentials` of configuration file.
```
Comment (markus-hentsch) [inline README.md:229]:
```suggestion
gx-credential-generator creates Gaia-X Credentials, which refer to [W3C Verifiable Credentials](https://www.w3.org/TR/vc-data-model-2.0/). Verifiable Credentials require a proof, e.g. a digital signature of the credential's issuer. Therefore some settings, e.g. a private key to sign, are required and defined in the section `Credentials` of the configuration file.
```

File: README.md [hunk 28]
```diff
@@ -196,11 +224,17 @@ cloud resources:
...
+
+### Enpoints to GXDCH services
+
+gx-credential-generator interacts with GXDCH service, e.g. to retreive credential for CSP's legal registration number or to assert compliance. To set GXDCH endpoint use section `gxdch`. For a list of available GXSCH endpoints refer to [Gaia-X Framework](https://docs.gaia-x.eu/framework/?tab=clearing-house)
```
Comment (markus-hentsch) [inline README.md:233]:
```suggestion
gx-credential-generator interacts with GXDCH service, e.g. to retreive credential for CSP's legal registration number or to assert compliance. To set the GXDCH endpoints use the options in the section `gxdch`. For a list of available GXSCH endpoints refer to [Gaia-X Framework](https://docs.gaia-x.eu/framework/?tab=clearing-house)
```

File: README.md [hunk 29]
```diff
@@ -1,182 +1,210 @@
...
+- OpenStack as an IaaS Offering
+- Kubernetes as a CaaS Offering
 
-- public VM Images
+Each description may consist of several Gaia-X Credentials, each of them attesting other properties.
+All credentials are bundled in a so called [Presentation](https://www.w3.org/TR/vc-data-model/#presentations-0) and send to [GXDCH Compliance Service](https://gaia-x.eu/gxdch/), which issues a Compliance Credential to certify Gaia-X compliance of given CSP and/or service.
 
-This then should be output as GX-compliant credential in JSON-LD for the Gaia-X
-catalogue.
+Gaia-X defines several level of compliance, each with a different trust level. gx-credential-generator supports the very basic level, called "Conformity" and is compliant Gaia-X Tagus release.
```
Comment (markus-hentsch) [inline README.md:32]:
```suggestion
Gaia-X defines several level of compliance, each with a different trust level. gx-credential-generator supports the very basic level, called "Conformity" and is compliant with the Gaia-X Tagus release.
```

**Review by markus-hentsch — CHANGES_REQUESTED** [review]

**Comment by anjastrunk** [comment]
@markus-hentsch Thanks for your carefully review. I see, English grammar and spelling is not one of my strength ;-) I applied your suggestions, please re-review.

File: README.md [hunk 30]
```diff
@@ -1,182 +1,212 @@
...
-So the GX Credential for a k8s aaS solution would list possible options and
-ranges: What k8s versions are supported, what max number of workers,
-flavors, etc.? What services are optionally delivered (and supported)
-by the provider?
+### Kubernetes (k8s)
 
-For KaaS, the option space really needs to be described.
-As of now, this can not be discovered, short of using external sources,
-like the IaaS GX Credential, the list of node images (osism minio), ...
+Comming soon.
```
Comment (markus-hentsch) [inline README.md:64]:
```suggestion
Coming soon.
```

File: README.md [hunk 31]
```diff
@@ -1,182 +1,212 @@
...
 
-## Compliance
+### Kubernetes (k8s)
 
-GX Credential Generator creates credentials compliant with the latest (
-2024/01/19) Credential Schema, which can be downloaded from
-the [GX registry](https://registry.lab.gaia-x.eu/v1/api/trusted-shape-registry/v1/shapes/trustframework).
-GX Credential Generator **does not** create W3C
-complaint [Verifiable Credentials](https://www.w3.org/TR/vc-data-model/).
+Comming soon!
```
Comment (markus-hentsch) [inline README.md:170]:
```suggestion
Coming soon!
```

File: README.md [hunk 32]
```diff
@@ -1,182 +1,212 @@
...
-  - `auth.user_domain_name`
-  - `auth.project_domain_name`
-  - `region_name`
+gx-credential-generator requires some configuration options. See [configuration](#configuration) section for more details.
 
-b. Generate Gaia-X Credentials
+#### 2. Run gx-credential-generator
 
-- To print OpenStack properties in JSON-LD
+Create Gaia-X Credential for CSP without specifying a configuration file. This implies the default path at `/etc/gx-credential-generator/config.yaml`, which must exist:
```
Comment (markus-hentsch) [inline README.md:91]:
```suggestion
Create Gaia-X Credential for a CSP without specifying a configuration file. This implies the default path at `/etc/gx-credential-generator/config.yaml`, which must exist:
```

File: README.md [hunk 33]
```diff
@@ -1,182 +1,212 @@
...
+Each Gaia-X Credential is serialized in [JSON-LD](https://json-ld.org/) and stored in a separate file prefixed as follows:
 
-  ```bash
-  ./gx-cred-generator.py k8s
-  ```
+- lp: Gaia-X Credential containing CSP's legal address and headquarter address
+- lrn: Gaia-X Credential for CSP's legal registration number issued by GXDCH Notary Service. gx-credential-generator reaches out to Notary Service by itself.
+- tandc: Gaia-X Terms and Conditions signed by CSP
+- vp_csp: Presentation of all Gaia-X Credentials to be sent to GXDCH Compliance Service to assert compliance
+- cs_csp: Compliance Credential for CSP as Gaia-X `LegalPerson` issued by GXDCH Compliance Service
```
Comment (markus-hentsch) [inline README.md:107]:
```suggestion
- lp: Gaia-X Credential containing the CSP's legal address and headquarter address
- lrn: Gaia-X Credential for the CSP's legal registration number issued by GXDCH Notary Service. gx-credential-generator reaches out to Notary Service by itself.
- tandc: Gaia-X Terms and Conditions signed by the CSP
- vp_csp: Presentation of all Gaia-X Credentials to be sent to GXDCH Compliance Service to assert compliance
- cs_csp: Compliance Credential for the CSP as Gaia-X `LegalPerson` issued by GXDCH Compliance Service
```

File: README.md [hunk 34]
```diff
@@ -1,182 +1,212 @@
...
-the
-simple SD validator script. Visit the `sd` directory and try to validate your
-generated SD. Find the examples in `sd` directory and do the validation as
-follows:
+As Gaia-X requires to define a provider for each published service offering, gx-credential-generator creates Gaia-X Credentials for the CSP at every run for an OpenStack cloud, too.
 
-Try to validate a minimal example against the latest GX shapes (feel free to
-remove some
-required attribute and check validation result):
+Each Gaia-X Credential is serialized in [JSON-LD](https://json-ld.org/) and stored in a separate file. Credentials for CSP correspond to the ones generated by the command `csp`. Credentials for OpenStack clouds are prefixed as follows:
```
Comment (markus-hentsch) [inline README.md:161]:
```suggestion
Each Gaia-X Credential is serialized in [JSON-LD](https://json-ld.org/) and stored in a separate file. Credentials for CSPs correspond to the ones generated by the command `csp`. Credentials for OpenStack clouds are prefixed as follows:
```

File: README.md [hunk 35]
```diff
@@ -1,182 +1,212 @@
...
 
-- To modify the airflow pipeline you have to touch the gaiax-pipeline.py file
-  inside the dags folder
+#### 2. Create configuration file
+
+gx-credential-generator requires some configuration options. See [configuration](#configuration) section for more details.
+
+#### 3. Run gx-credential-generator
+
+The command to run gx-credential-generator for OpenStack cloud is similar to the one to run gx-credential-generator for CSP. Also arguments `--config`, `--out-dir` and `--auto-sign` are available and act like for CSP.
```
Comment (markus-hentsch) [inline README.md:149]:
```suggestion
The command to run gx-credential-generator for OpenStack clouds is similar to the one to run gx-credential-generator for a CSP. The arguments `--config`, `--out-dir` and `--auto-sign` are also available and act like for the CSP command.
```

File: README.md [hunk 36]
```diff
@@ -196,11 +226,17 @@ cloud resources:
...
 ```
 
-### Cryptographical Material
+### Prerequisites to create and sign Gaia-X Credential
+
+gx-credential-generator creates Gaia-X Credentials, which refer to [W3C Verifiable Credentials](https://www.w3.org/TR/vc-data-model-2.0/). Verifiable Credentials require a proof, e.g. a digital signature of the credential's issuer. Therefore some settings, e.g. a private key to sign, are required and defined in the section `Credentials` of the configuration file.
+
+### Enpoints to GXDCH services
+
+gx-credential-generator interacts with GXDCH service, e.g. to retreive credential for CSP's legal registration number or to assert compliance. To set the GXDCH endpoints use the options in the section `gxdch`. For a list of available GXSCH endpoints refer to [Gaia-X Framework](https://docs.gaia-x.eu/framework/?tab=clearing-house)
```
Comment (markus-hentsch) [inline README.md:235]:
```suggestion
gx-credential-generator interacts with GXDCH service, e.g. to retrieve a credential for a CSP's legal registration number or to assert compliance. To set the GXDCH endpoints use the options in the section `gxdch`. For a list of available GXSCH endpoints refer to [Gaia-X Framework](https://docs.gaia-x.eu/framework/?tab=clearing-house)
```

**Review by markus-hentsch — CHANGES_REQUESTED** [review]
Just added some small adjustments that I either missed last time or that have changed again after the wave of commits.

**Comment by anjastrunk** [comment]
@markus-hentsch I applies all your suggestions. Thanks a lot for your contribution.

File: README.md [hunk 37]
```diff
@@ -1,182 +1,212 @@
...
 
    ```bash
    pip install -r requirements.txt
    ```
 
-### OpenStack
+### Cloud Service Provider
 
-a. Create `clouds.yaml` configuration file
+### 1. Create configuration file
```
Comment (markus-hentsch) [inline README.md:85]:
```suggestion
#### 1. Create configuration file
```

**Review by markus-hentsch — CHANGES_REQUESTED** [review]
Had one last look over the rendered version and noticed one heading being on the wrong level. Other than, that looks good now.

**Comment by anjastrunk** [comment]
> Had one last look over the rendered version and noticed one heading being on the wrong level. Other than, that looks good now.

Done.

**Review by markus-hentsch — APPROVED** [review]
This looks good to me now. Thanks for the adjustments!

**Pull request merged**